cpp/cmake_modules/FindSnappy.cmake

```diff
@@ -15,25 +15,33 @@
 # specific language governing permissions and limitations
 # under the License.
 
+set(SNAPPY_STATIC_LIB_SUFFIX "${SNAPPY_MSVC_STATIC_LIB_SUFFIX}${CMAKE_STATIC_LIBRARY_SUFFIX}")
+
+if(ARROW_SNAPPY_USE_SHARED)
+  set(SNAPPY_LIB_NAMES snappy)
+else()
+  set(SNAPPY_LIB_NAMES "${CMAKE_STATIC_LIBRARY_PREFIX}snappy${SNAPPY_STATIC_LIB_SUFFIX}" snappy)
+endif()
+
 if(Snappy_ROOT)
   find_library(Snappy_LIB
-               NAMES snappy
+               NAMES ${SNAPPY_LIB_NAMES}
                PATHS ${Snappy_ROOT}
-               PATH_SUFFIXES ${LIB_PATH_SUFFIXES}
+               PATH_SUFFIXES ${ARROW_LIBRARY_PATH_SUFFIXES}
                NO_DEFAULT_PATH)
   find_path(Snappy_INCLUDE_DIR
             NAMES snappy.h
             PATHS ${Snappy_ROOT}
             NO_DEFAULT_PATH
-            PATH_SUFFIXES ${INCLUDE_PATH_SUFFIXES})
+            PATH_SUFFIXES ${ARROW_INCLUDE_PATH_SUFFIXES})
 else()
-  find_library(Snappy_LIB NAMES snappy)
-  find_path(Snappy_INCLUDE_DIR NAMES snappy.h PATH_SUFFIXES ${INCLUDE_PATH_SUFFIXES})
+  find_library(Snappy_LIB NAMES ${SNAPPY_LIB_NAMES} HINTS "${CMAKE_ROOT}/Modules/")
```

Comment:
Could you explain why the `HINTS` is needed?
I don't think that `libsnappy.so` exists in `${CMAKE_ROOT}/Modules/`.

Reply:
I don't remember why I added this. It was probably a conversion error.